python/lsst/drp/tasks/assemble_coadd.py

```diff
@@ -0,0 +1,1777 @@
+                radius = int(rec["radius"].asArcseconds()/plateScale)   # convert to pixels
+                spans = afwGeom.SpanSet.fromShape(radius, offset=center)
+            else:
+                self.log.warning("Unexpected region type %s at %s", rec["type"], center)
+                continue
+            spans.clippedTo(mask.getBBox()).setMask(mask, self.brightObjectBitmask)
+
+    def setInexactPsf(self, mask):
+        """Set INEXACT_PSF mask plane.
+
+        If any of the input images isn't represented in the coadd (due to
+        clipped pixels or chip gaps), the `CoaddPsf` will be inexact. Flag
+        these pixels.
+
+        Parameters
+        ----------
+        mask : `lsst.afw.image.Mask`
+            Coadded exposure's mask, modified in-place.
+        """
+        mask.addMaskPlane("INEXACT_PSF")
+        inexactPsf = mask.getPlaneBitMask("INEXACT_PSF")
+        sensorEdge = mask.getPlaneBitMask("SENSOR_EDGE")  # chip edges (so PSF is discontinuous)
+        clipped = mask.getPlaneBitMask("CLIPPED")  # pixels clipped from coadd
+        rejected = mask.getPlaneBitMask("REJECTED")  # pixels rejected from coadd due to masks
+        array = mask.getArray()
+        selected = array & (sensorEdge | clipped | rejected) > 0
+        array[selected] |= inexactPsf
+
+    @staticmethod
+    def _subBBoxIter(bbox, subregionSize):
+        """Iterate over subregions of a bbox.
+
+        Parameters
+        ----------
+        bbox : `lsst.geom.Box2I`
+            Bounding box over which to iterate.
+        subregionSize : `lsst.geom.Extent2I`
+            Size of sub-bboxes.
+
+        Yields
+        ------
+        subBBox : `lsst.geom.Box2I`
+            Next sub-bounding box of size ``subregionSize`` or smaller; each
+            ``subBBox`` is contained within ``bbox``, so it may be smaller than
+            ``subregionSize`` at the edges of ``bbox``, but it will never be
+            empty.
+
+        Raises
+        ------
+        RuntimeError
+            Raised if any of the following occur:
+            - The given bbox is empty.
+            - The subregionSize is 0.
+        """
+        if bbox.isEmpty():
+            raise RuntimeError("bbox %s is empty" % (bbox,))
+        if subregionSize[0] < 1 or subregionSize[1] < 1:
+            raise RuntimeError("subregionSize %s must be nonzero" % (subregionSize,))
+
+        for rowShift in range(0, bbox.getHeight(), subregionSize[1]):
+            for colShift in range(0, bbox.getWidth(), subregionSize[0]):
+                subBBox = geom.Box2I(bbox.getMin() + geom.Extent2I(colShift, rowShift), subregionSize)
+                subBBox.clip(bbox)
+                if subBBox.isEmpty():
+                    raise RuntimeError("Bug: empty bbox! bbox=%s, subregionSize=%s, "
+                                       "colShift=%s, rowShift=%s" %
+                                       (bbox, subregionSize, colShift, rowShift))
+                yield subBBox
```

Comment:
This is in `lsst.pipe.tasks.coaddBase`, so it can be removed as a static method of this class.